### PR TITLE
PR #23 (add/loadConfig)

### PR DESCRIPTION
**Pull request by rgadon107**
@hellofromtonya I'm scratching my head. When I call the assert in the test method `test_should_store_config_in_the_store_given_valid_key_and_config`, the API function `loadConfig` returns the error message: 

>Failed asserting that null is identical to 'foo'.

When I use `var_dump()` to inspect `$config`, and `_the_store()` in the test method, I get a value back. But when I inspect `loadConfig( $store_key, $config )`, `var_dump()` returns a value of NULL. 

Why does `loadConfig()` return a value of NULL when I pass the function it's expected parameters?

### CODE REVIEW

**Comment by rgadon107** [comment]
@hellofromtonya I'm working on the second unit test for this branch ( `test_should_throw_exception_when_store_key_is_empty` ; See commit 
[4ce2c06](url) ).  

Composer throws the error message: 

>Failed asserting that exception message '' contains 'Configuration for [] does not exist in the ConfigStore'.

I know that when I mock `_the_store` and pass that function an empty store key, I can return an Exception error. But I can't properly assert the ExceptionMessage.  Am I doing something wrong with `sprintf()`? Or am I failing to set up the test properly?

**Comment by hellofromtonya** [comment]
@rgadon107 Take a look at commit https://github.com/rgadon107/cornerstone/pull/23/commits/56afd25cb9780fa17dde68a7f4573432da6d352b.  To mock throw an Exception, it takes a bit more logic.

**Comment by rgadon107** [comment]
@hellofromtonya Sometimes a blatant error is sitting right out in front creating all kinds of crazy & unexpected behavior on the CLI. And I didn't even see it until I left the screen for a couple of hours and returned to the problem of unit test #3. The 3 edge cases are now addressed. I'm ready for your code review of this branch.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/loadConfig.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	public function test_should_store_config_in_the_store_given_valid_key_and_config() {
+		$config = [
+			'aaa' => 'bbb',
+			'ccc' => 'ddd',
+		];
+		Monkey\Functions\expect( 'KnowTheCode\ConfigStore\_the_store' )
+			->once()
+			->with( 'foo', $config )
+			->andReturn( true );
+
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/loadConfig.php]: Be consistent with the code formatting.  Notice here you have 1 empty line but below there's none.  Be consistent.  Pick one style and stick with it.

**Comment by rgadon107** [comment]
@hellofromtonya Per your earlier comment about file formatting, I fixed the placement of the `assert` in [a974b6e](a974b6e). 

I'm stuck on writing the 2nd and 3rd integration test (see commit [953fe00](953fe00) ).  I get a failure message on each of the 2 tests: 

>1) spiralWebDb\centralHub\Tests\Integration\ConfigStore\Tests_LoadConfig::test_should_throw_exception_when_given_empty_key_and_valid_config

>Failed asserting that exception of type "Exception" is thrown.

>2) spiralWebDb\centralHub\Tests\Integration\ConfigStore\Tests_LoadConfig::test_should_throw_exception_when_given_valid_key_and_empty_config

>Failed asserting that exception of type "Exception" is thrown. 

I need to take a break from this. Not sure on what to do next. 

When I `var_dump( loadConfig( '', $config) );` to inspect the function and run the integration test, the function dumps out everything that's stored in `_the_store()` from the cornerstone project. So clearly data _is stored_ in the store. Now sure how to isolate things here.

**Comment by hellofromtonya** [comment]
>I'm stuck on writing the 2nd and 3rd integration test (see commit 953fe00 ). I get a failure message on each of the 2 tests:

For the 2nd integration test, look at the code for `_the_store()`.  What happens when there's no store key?

```php
	// Bail out if no store key is given.
	if ( ! $store_key ) {
		return $config_store;
	}
```

It bails out and returns everything in the store.  We need this behavior for the function `getAllKeys()`.  But let's put on our thinking caps:

1. `getAllKeys()` uses all the defaults and just calls `_the_store()`, meaning it would not have a config to store.
2. But what if a store key was not given but a config to store was?  Whoopsie, that looks like a new error condition that we've not addressed previously.  

Bummer.  Let's fix that.  I'm working on a commit for it now.

**Comment by hellofromtonya** [comment]
@rgadon107 As I noted above, we needed to modify `_the_store()`.  I did that in PR #25, which has been merged into `development` and this PR.  Those changes in `_the_store()` fixed the errors you were having above, i.e. well after I massaged the messaging.  And it lets `getAllKeys()` work too.

Look it over.  Ask any questions.

**Comment by rgadon107** [comment]
@hellofromtonya Per your earlier comment about file formatting, I fixed the placement of the `assert` in [a974b6e](a974b6e). 

I'm stuck on writing the 2nd and 3rd integration test (see commit [953fe00](953fe00) ).  I get a failure message on each of the 2 tests: 

>1) spiralWebDb\centralHub\Tests\Integration\ConfigStore\Tests_LoadConfig::test_should_throw_exception_when_given_empty_key_and_valid_config

>Failed asserting that exception of type "Exception" is thrown.

>2) spiralWebDb\centralHub\Tests\Integration\ConfigStore\Tests_LoadConfig::test_should_throw_exception_when_given_valid_key_and_empty_config

>Failed asserting that exception of type "Exception" is thrown. 

I need to take a break from this. Not sure on what to do next. 

When I `var_dump( loadConfig( '', $config) );` to inspect the function and run the integration test, the function dumps out everything that's stored in `_the_store()` from the cornerstone project. So clearly data _is stored_ in the store. Now sure how to isolate things here.
